packages/strapi-plugin-upload/admin/src/utils/createFileToDownloadName.js

```diff
@@ -1,4 +1,8 @@
-const createFileToDownloadName = ({ file: { ext }, fileInfo: { name } }) => {
+import { get } from 'lodash';
+
+const createFileToDownloadName = ({ file, fileInfo: { name } }) => {
+  const ext = get(file, 'ext', '');
+
   return `${name.replace(ext, '')}${ext}`;
```

Comment:
In case the extension is also present in the name (ex: `my.txtfile.txt`), I think this line would not do want expected.
What about:
```
return name.endsWith(ext)
  ? name
  : `${name}${ext}`;
```